src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/NotificationsService.cs

```diff
@@ -26,27 +26,23 @@ public class NotificationsService : INotificationsService
 
         public async Task NotifySubscribersIfApplicable(params Guid[] releaseIds)
         {
-            var releasesToNotify = _context.Releases
-                .Include(r => r.ReleaseStatuses)
-                .Where(r => releaseIds.Contains(r.Id))
-                .ToList()
-                .Where(r => r.NotifySubscribers)
-                .ToList();
+            var releasesToNotify = await _context.Releases
+                .Where(r => releaseIds.Contains(r.Id) && r.NotifySubscribers && r.NotifiedOn == null)
+                .ToListAsync();
+
             var messages = await releasesToNotify
                 .ToAsyncEnumerable()
                 .SelectAwait(async release => await BuildPublicationNotificationMessage(release))
                 .ToListAsync();
+
             if (messages.Count > 0)
             {
                 await _storageQueueService.AddMessages(ReleaseNotificationQueue, messages);
                 releasesToNotify
                     .ForEach(release =>
                     {
-                        var status = release.ReleaseStatuses
-                            .OrderBy(rs => rs.Created)
-                            .Last();
-                        _context.Update(status);
-                        status.NotifiedOn = DateTime.UtcNow;
+                        _context.Update(release);
```

Comment:
Maybe `_context.Releases.Update(release);` for the type safety?  I know this code isn't 100% new but I look to point this out if I see any occurrences in PRs now where we're just using the blanked "Update" or "Add" / "AddRange" etc methods rather than the collection-specific ones that provide a bit more safety.

Reply:
Ah yep, made that change now, thanks. 🙂

Reply:
Ta